src/main/java/com/dao/homework/dao/jdbc/CarDaoJdbcImpl.java

```diff
@@ -0,0 +1,203 @@
+                    + carId + " from DB ", e);
+        }
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateQuery = "UPDATE cars SET manufacturer_id = ?, cars_model = ? "
+                + "WHERE cars_id = ? AND deleted = false";
+        String deleteDriversQuery = "DELETE FROM drivers_cars WHERE car_id = ?";
+        String insertDriversQuery = "INSERT INTO drivers_cars(driver_id, car_id) VALUES (?, ?)";
+        Long carId = car.getId();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateCarStmt = connection.prepareStatement(updateQuery);
+                PreparedStatement deleteDriversStmt =
+                        connection.prepareStatement(deleteDriversQuery);
+                PreparedStatement insertDriversStmt =
+                        connection.prepareStatement(insertDriversQuery)) {
+            updateCarStmt.setLong(1, car.getManufacturer().getId());
+            updateCarStmt.setString(2, car.getModel());
+            updateCarStmt.setLong(3, carId);
+            updateCarStmt.executeUpdate();
+            deleteDriversStmt.setLong(1, carId);
+            deleteDriversStmt.executeUpdate();
+            for (Driver driver : car.getDrivers()) {
+                insertDriversStmt.setLong(1, driver.getId());
+                insertDriversStmt.setLong(2, carId);
+                insertDriversStmt.executeUpdate();
+            }
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("We can`t update this car:" + car, e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long carsId) {
+        String query = "UPDATE " + TABLE_NAME + " SET " + DELETED
+                + "= ?  WHERE cars_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setBoolean(1, true);
+            statement.setLong(2, carsId);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t delete data with id:"
+                    + carsId + " from DB ", e);
+        }
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String query = "SELECT d.drivers_id, d.drivers_name, d.license_number,\n"
+                + "c.cars_id,c.cars_model,drivers_id,m.manufacturer_id,"
+                + "m.manufacturer_name,m.manufacturer_country\n"
+                + "FROM cars c\n"
+                + "\tLEFT JOIN drivers_cars dr_ca\n"
+                + "    ON c.cars_id = dr_ca.car_id\n"
+                + "    LEFT JOIN drivers d\n"
+                + "    ON c.cars_id = dr_ca.car_id\n"
+                + "    LEFT JOIN manufacturer m\n"
+                + "    ON c.manufacturer_id = m.manufacturer_id";
```

Comment:
please check every item on the common mistakes

Reply:
I`m not sure than i understans what do you mean right